### ISSUE TITLE
synthesize() silently produces all-NA output and changes dtypes when input contains heavy NaN columns

### ISSUE THREAD

**Issue by talhabektas** [dataxid/dataxid-python#69]
## Summary
When `dataxid.synthesize()` is called on a DataFrame containing heavy missing values (including one fully-NaN column), the SDK reports `SUCCESS`, returns the requested number of rows, but **every cell in the output is `<NA>`** and the dtype of every column is silently changed to `string[python]` — including a column that was originally `float64`. There is no warning or error.

## Environment
- OS: Ubuntu 22.04 (Google Colab, L4 GPU)
- Python: 3.12
- dataxid: 0.1.0
- pandas: 2.2.2

## Steps to Reproduce
```python
import dataxid
import pandas as pd

dataxid.api_key = "dx_..."

df = pd.DataFrame({
    "a": [1, None, None, 4, None, None, 7, None, None, 10],   # numeric with NaN
    "b": [None]*10,                                            # all NaN
    "c": ["x", None, "y", None, "z", None, "x", None, "y", None],  # categorical with NaN
})
print("Input dtypes:")
print(df.dtypes)
# a    float64
# b     object
# c     object

out = dataxid.synthesize(data=df, n_samples=20)
print(f"\nOutput shape: {out.shape}")     # (20, 3) — looks fine
print(f"Output dtypes:")
print(out.dtypes)
# a    string[python]   ← changed from float64
# b    string[python]
# c    string[python]
print(f"\nOutput sample:")
print(out.head(5))
#      a     b     c
# 0  <NA>  <NA>  <NA>
# 1  <NA>  <NA>  <NA>
# ...
```

## Expected Behavior
Either:
- Raise a clear error / warning that column `b` is fully NaN and cannot be modeled, before training begins, **or**
- Train successfully and produce synthetic values that respect the input dtypes (numeric column should stay numeric) and at least partially fill non-NaN values for columns `a` and `c`.

In neither case should the SDK silently:
- Change a `float64` column to `string[python]`
- Return all `<NA>` for columns that had real values in the input

## Actual Behavior
- No error, no warning, status = `SUCCESS`
- Training reports `train_loss=1.4459, val_loss=1.2620, early_stopped=True` (suspiciously fast — 6 epochs)
- All 20×3 = 60 cells in the output are `<NA>`
- All dtypes silently coerced to `string[python]`

This is effectively **silent data loss**. A user piping `synthesize()` output into a downstream ML pipeline would see a 20-row DataFrame and assume it succeeded, only to discover later that there's no usable data.

## Severity
**Critical (10)** — silent data loss with `SUCCESS` status. Worse than a crash because failures are invisible.

## Root Cause Hypothesis
The encoder appears to mark every column as `TABULAR_CATEGORICAL` (visible in the training logs) when the column contains heavy NaN values, including the numeric column `a`. During decoding, the categorical decoder cannot recover original values because its vocabulary collapses to NaN. The result is silently materialized as `<NA>` strings without surfacing a warning.

## Fix Suggestion
1. **At analyze time**, detect columns where >50% of values are NaN and either skip them with a clear log warning or raise an actionable error:
```python
nan_ratio = df[col].isna().mean()
if nan_ratio == 1.0:
    raise ValueError(f"Column {col!r} is entirely NaN and cannot be modeled.")
if nan_ratio > 0.5:
    warnings.warn(f"Column {col!r} is {nan_ratio:.0%} NaN; synthesis quality will be degraded.")
```
2. **Preserve input dtypes** through the round trip; if the encoder coerces numeric → categorical due to NaN, decode back to the original numeric dtype.
3. **Validate output before returning**: if >50% of output cells are NA, surface a warning so users notice.

## Additional Notes
- Reproduced on Colab (Linux). The bug is in the encoding/decoding pipeline.
- Discovered as part of LLM-assisted testing for SWE422 course at Sakarya University.

**Comment by BurakUlver** [comment]
An all-null column stays null because there's nothing to learn. Ten rows fail training with training_diverged.
